generate-docker-image-tags/action.yml

```diff
@@ -62,14 +62,17 @@ runs:
 
         if [ "$GITHUB_EVENT_NAME" == "pull_request" ]
         then
-          PR_NUMBER=${{ github.event.number }}
-        else
+          PR_NUMBER="${{ github.event.number }}"
+        elif [ "$GITHUB_EVENT_NAME" == "push" ]
+        then
           # Workflow was triggered by a "push" event.
-          PR_NUMBER=$(
+          PR_NUMBER='$(
             echo "${{ github.event.head_commit.message }}" \
             | grep -o "(#[0-9]*)" \
-            | sed 's/[#()]//g'
-          )
+            | sed "s/[#()]//g"
+          )'
```

Comment:
The output of the GItHub Actions runs look right but I'm confused why the `'` quotes actually work.

When I run something similar in a `bash` shell (v3.2), I get
```bash
$ x='$(
> pwd
> )'
$ echo $x
$( pwd )
```
compared to
```bash
$ x=$(
> pwd
> )
$ echo $x
/Users/mikhail
```

Reply:
Ah, I figured out why running this code _without_ the quotes fail!

The issue is that GitHub Actions runs each `bash` workflow step using the following command:
```shell
/usr/bin/bash --noprofile --norc -e -o pipefail {0}
```
This means that if a `|` operation fails, the whole script fails. The culprit then is `grep` which returns exit code `1` if there are no matches which happens on all non-PR pushes.

Reply:
Granted, I still have no clue why using `'` quotes work. 😅

Reply:
Thanks for discovering the root issue @Mandrenkov ! I've pushed a fix to this PR, test results enclosed:
https://github.com/XanaduAI/doppler-radar/actions/runs/3446634622
https://github.com/XanaduAI/doppler-radar/actions/runs/3446638485
https://github.com/XanaduAI/doppler-radar/actions/runs/3446645003

Reply:
Looks 👌!